chain/syncer.go

```diff
@@ -99,10 +102,14 @@ type Syncer struct {
 	chainStore syncerChainReaderWriter
 	// Provides message collections given cids
 	messageProvider MessageProvider
+
+	statusMu sync.Mutex
```

Comment:
The clock should be reentrant, so move it above the mutex to avoid implying that it's protected by it.

If the status value is to be mutated in place (as currently in endRun) then the mutex is most appropriate.